thirdparty/chart-auth-svc/templates/deployment.yaml

```diff
@@ -0,0 +1,72 @@
+{{- if .Values.enabled -}}
+apiVersion: {{ template "apiVersion.Deployment" . }}
+kind: Deployment
+metadata:
+  name: {{ include "auth-svc.name" . }}
+  labels:
+    {{- include "auth-svc.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+  selector:
+    matchLabels:
+      {{- include "auth-svc.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "auth-svc.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          command: {{ .Values.image.command }}
+          {{- if .Values.readinessProbe.enabled }}
+          readinessProbe:
+            httpGet:
+              path: {{ .Values.readinessProbe.httpGet.path }}
+              port: {{ .Values.service.ports.internalPort }}
+            initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+          {{- end }}
+          {{- if .Values.livenessProbe.enabled }}
+          livenessProbe:
+            httpGet:
+              path: {{ .Values.livenessProbe.httpGet.path }}
+              port: {{ .Values.service.ports.internalPort }}
+            initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+          {{- end }}
+          volumeMounts:
+            - name: auth-svc-config-volume
+              mountPath: /opt/auth-service/config/production.json
+              subPath: production.json
+          env:
+          {{- range $envItem := .Values.env }}
+            - name: {{ $envItem.name }}
+              value: {{ $envItem.value }}
+          {{- end }}
+      volumes:
+        - name: auth-svc-config-volume
+          configMap:
+            name: auth-svc-config
+            items:
+              {{- range $k, $v := index .Values.config }}
```

Comment:
Any reason that this generally loads all config files, but the above volumeMounts a specific file `production.json`?

If you are not planning to do something like that, I would just fetch the "production.json".

Reply:
Sure, I can do that. 

I thought this approach would be more generic, in case we wanted to specify more than one file and be able to switch between them easily. That is similar to the approach we use when switching development environments with convict - see https://github.com/mojaloop/auth-service/tree/master/config for example